tests/http/test_proxyagent.py

```diff
@@ -240,7 +241,10 @@ def test_http_request_via_proxy(self):
         body = self.successResultOf(treq.content(resp))
         self.assertEqual(body, b"result")
 
-    @patch.dict(os.environ, {"https_proxy": "proxy.com", "no_proxy": "unused.com"})
+    @patch.dict(
+        os.environ,
+        {"https_proxy": "bob:pinkponies@proxy.com", "no_proxy": "unused.com"},
+    )
```

Comment:
Can you make a new test please? So that we test both paths.

Reply:
Sure. I'll try to de-duplicate the code a bit between them.